### PR TITLE
Style inspector

### PR DESCRIPTION
**Pull request by MarchingCube**
## What does the pull request do?

https://user-images.githubusercontent.com/2588062/108751969-2d684980-7543-11eb-9105-f2ae468fd3f4.mp4

Update: Implemented style snapshotting (snapshot current styles to easily inspect state). To simplify inspection I've added a way of copying property names and values.

https://user-images.githubusercontent.com/2588062/109233372-12f3d180-77ca-11eb-88bb-9c80dae2be7c.mp4

## What is the current behavior?

It is fairly hard to tell why given control has given property value especially with several style layers are involved.

## What is the updated/expected behavior with this PR?

User can check styles on a given control, we are also listing set properties and values.
Properties that are inactive (due to another style overriding it) are marked too.
Pseudo classes can also be enabled manually to hunt down the most elusive pointerover or pressed styles.

## Checklist

- [ ] Added unit tests (if possible)?
- [ ] Added XML documentation to any related classes?
- [ ] Consider submitting a PR to https://github.com/AvaloniaUI/Avaloniaui.net with user documentation

### CODE REVIEW

File: src/Avalonia.Styling/StyledElement.cs [hunk 1]
```diff
@@ -16,6 +16,24 @@
 
 namespace Avalonia
 {
+    public class StyleDiagnostics
```
Comment (grokys) [inline src/Avalonia.Styling/StyledElement.cs:19]: Should probably move these to the `Avalonia.Diagnostics` namespace.

**Comment by grokys** [comment]
This is excellent! And surprisingly simple. What's left to do?

File: src/Avalonia.Styling/StyledElement.cs (hunk 1 above)
Comment (MarchingCube) [reply src/Avalonia.Styling/StyledElement.cs]: Yep, done.

**Comment by MarchingCube** [comment]
@grokys I need to cleanup a bit and implement a few quality of life features (snapshots, context menus). But this is done now - publishing :)

**Review by grokys — APPROVED** [review]
LGTM! Lets get this merged so everyone can experience its goodness!

**Comment by PieroCastillo** [comment]
what means the crossed out words?

**Comment by maxkatz6** [comment]
@PieroCastillo overridden value by style-setter/animation with higher priority.

**Pull request merged**